src/chainstate/stacks/boot/pox_2_tests.rs

```diff
@@ -0,0 +1,423 @@
+
+    assert_eq!(tip.block_height, 13 + EMPTY_SORTITIONS as u64);
+
+    let alice_lockup = make_pox_2_lockup(
+        &alice,
+        1,
+        512 * POX_THRESHOLD_STEPS_USTX,
+        AddressHashMode::SerializeP2PKH,
+        key_to_stacks_addr(&alice).bytes,
+        12,
+        tip.block_height,
+    );
+    peer.tenure_with_txs(&[alice_lockup], &mut coinbase_nonce);
+
+    // Alice locked half her balance in PoX 2
+    let alice_balance = get_balance(&mut peer, &key_to_stacks_addr(&alice).into());
+    assert_eq!(alice_balance, 512 * POX_THRESHOLD_STEPS_USTX);
+
+    // now, let's roll the chain forward until Alice *would* have unlocked in v1 anyways.
+    //  that's block height 31, so play 27 empty blocks
+
+    for _i in 0..17 {
+        peer.tenure_with_txs(&[], &mut coinbase_nonce);
+        // at this point, alice's balance should always include this half lockup
+        assert_eq!(alice_balance, 512 * POX_THRESHOLD_STEPS_USTX);
+    }
+
+    let tip =
+        SortitionDB::get_canonical_burn_chain_tip(&peer.sortdb.as_ref().unwrap().conn()).unwrap();
+    assert_eq!(tip.block_height, 31 + EMPTY_SORTITIONS as u64);
+    let alice_lockup = make_pox_lockup(
+        &alice,
+        2,
+        512 * POX_THRESHOLD_STEPS_USTX,
+        AddressHashMode::SerializeP2PKH,
+        key_to_stacks_addr(&alice).bytes,
+        12,
+        tip.block_height,
+    );
+    peer.tenure_with_txs(&[alice_lockup], &mut coinbase_nonce);
+
+    assert_eq!(alice_balance, 512 * POX_THRESHOLD_STEPS_USTX);
+
+    // now let's check some tx receipts
+
+    let alice_address = key_to_stacks_addr(&alice);
+    let bob_address = key_to_stacks_addr(&bob);
+    let blocks = observer.get_blocks();
+
+    let mut alice_txs = HashMap::new();
+    let mut bob_txs = HashMap::new();
+
+    eprintln!("Alice addr: {}", alice_address);
+    eprintln!("Bob addr: {}", bob_address);
+
+    let mut tested_charlie = false;
+
+    for b in blocks.into_iter() {
+        for r in b.receipts.into_iter() {
+            if let TransactionOrigin::Stacks(ref t) = r.transaction {
+                let addr = t.auth.origin().address_testnet();
+                eprintln!("TX addr: {}", addr);
+                if addr == alice_address {
+                    alice_txs.insert(t.auth.get_origin_nonce(), r);
+                } else if addr == bob_address {
+                    bob_txs.insert(t.auth.get_origin_nonce(), r);
+                } else if addr == key_to_stacks_addr(&charlie) {
+                    assert!(
+                        r.execution_cost != ExecutionCost::zero(),
+                        "Execution cost is not zero!"
+                    );
+                    tested_charlie = true;
+                }
+            }
+        }
+    }
+
+    assert!(tested_charlie, "Charlie TX must be tested");
+    assert_eq!(alice_txs.len(), 3, "Alice should have 3 confirmed txs");
+    assert_eq!(bob_txs.len(), 2, "Bob should have 2 confirmed txs");
+
+    assert!(
```

Comment:
For each assert in this section, where you are checking transactions, can you put a quick comment about *why* we expect these things to happen?